### ISSUE TITLE
[MS] Create container level rules for primary input data in growing datasets

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11130]
**Impact of the new feature**
MSTransferor

**Is your feature request related to a problem? Please describe.**
A sub-issue of: https://github.com/dmwm/WMCore/issues/11048

As we commission growing datasets/workflows in the system, we need to ensure that input blocks freshly created will have replica(s) in disk for re-processing.

The current design of MSTransferor does not allow it, because the workflow is processed by the service only once, blocks are locked and distributed on disk and the workflow advances to `staging`, thus never returning to MSTransferor. 

**Describe the solution you'd like**
Based on the workflow attribute `OpenRunningTimeout`, give a different treatment to the primary dataset, such that the whole container gets a rule locking it on disk to be re-processed, instead of the block based data placement.

I'd say that OpenRunningTimeout larger than a week should trigger this feature, anything smaller than that - or not defined - should follow the standard block-based input data placement. We better make this parameter configurable as well.

Question: should we lock the whole container in only one RSE, out of the possible RSEs the workflow has been assigned to? Or should we have a number of copies > 1 whenever possible (if so, how many)?

**Describe alternatives you've considered**
It just came to my mind that we might struggle to move this workflow forward in MSMonitor service, since it will have only 1 rule and it should be only satisfied if we spend many hours (maybe days) without generating new blocks. Something that might require another issue/change to be considered.

**Additional context**
None

**Comment by amaltaro** [comment]
If anyone has an input on how we should deal with input data placement in such workflows, I'd be glad to hear.
@haozturk you might have some input to provide.

**Comment by amaltaro** [comment]
And here are two concerns raised during the P&R Dev meeting:
1) making a container rule against a single RSE: it could be that eventually that RSE runs out of space and we are not able to transfer more input blocks in there.
2) making a container rule against multiple RSEs (with grouping=ALL): it could be that eventually an RSE runs out of space, and Rucio decides to transfer (move!) everything to another RSE that has enough volume for everything (thus, changing block location).

Perhaps the best way to deal with this would be to always target multiple RSEs and define grouping=DATASET, such that blocks can be scattered in multiple RSEs. Whether or not we add copies > 1 is still to be decided as well.

@ericvaandering and Felipe, that I do not know his GH user name.
